### PR TITLE
[filter/SNPE] Add tensor_filter sub-plugin for SNPE

### LINKED ISSUE #1872: Tensor_Filter::SNPE?
In addition to Project SN, SNPE is to be considered later for Qualcomm devices in the open.

Mid priority.

### PR DESCRIPTION
**Pull request by anyj0527**
- Developer should define SNPE_ROOT (dir path to Qualcomm Neural Processing SDK) which contains SNPE headers and libraries.
- Currently, this supports only CPU runtime on linux-x86_64

sample pipeline :
```
$ gst-launch-1.0 videotestsrc num-buffers=5 ! videoconvert ! videoscale !  video/x-raw,width=299,height=299,format=RGB,framerate=30/1 ! \
tensor_converter ! tensor_transform mode=arithmetic option=typecast:float32,add:-127.5,div:127.5 ! tensor_filter framework=snpe model=./inception_v3.dlc ! fakesink
```

sample app :
https://github.com/anyj0527/nnstreamer-example/tree/example/snpe/image-classification/native/example_image_classification_snpe

Fixes #1872

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/meson.build [hunk 1]
```diff
@@ -459,3 +459,63 @@ if get_option('enable-mediapipe')
     install_dir: filter_subplugin_install_dir
   )
 endif
+
+if get_option('enable-snpe')
+  if host_machine.system() != 'linux' and host_machine.cpu_family() != 'x86_64'
```
Comment (helloahn) [inline ext/nnstreamer/tensor_filter/meson.build:464]: Q: does SNPE support only linux x86_64?

File: ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+}
+
+/** @brief Destruct the subplugin */
+void snpe_subplugin::fini_filter_snpe (void)
+{
+  assert (registeredRepresentation != nullptr);
+  tensor_filter_subplugin::unregister_subplugin (registeredRepresentation);
+}
+
+extern "C" {
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc:283]: Why this is required?

Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc:283]: This statement is for nnstreamer-native-api

File: ext/nnstreamer/tensor_filter/meson.build (hunk 1 above)
Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/meson.build:464]: No, SNPE also supports arm architecture and Android. But the their development environment is limited Ubuntu.

File: ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc (hunk 2 above)
Comment (dongju-chae) [reply ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc:283]: You already declare `extern "C"` in line 41. So, I think this is not necessary.

Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc:283]: Thanks! I think you're right. I'll modify this part.

File: ext/nnstreamer/tensor_filter/meson.build [hunk 3]
```diff
@@ -459,3 +459,63 @@ if get_option('enable-mediapipe')
     install_dir: filter_subplugin_install_dir
   )
 endif
+
+if get_option('enable-snpe')
```
Comment (myungjoo) [inline ext/nnstreamer/tensor_filter/meson.build:463]: Use the new "feature" meson feature. Rebase to the recent master HEAD, then you will see it (@zhoonit has updated meson.build accordingly)

Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/meson.build]: @myungjoo I updated the meson.build to enable this snpe feature w.r.t. the existence of environment variable `$SNPE_ROOT`. Please check.

File: ext/nnstreamer/tensor_filter/meson.build (hunk 1 above)
Comment (anyj0527) [reply ext/nnstreamer/tensor_filter/meson.build]: @helloahn @myungjoo I updated the meson.build not to restrict the host_machine cause SNPE SDK also provides libraries for other host.
I'll deal with this in later PRs.

**Comment by myungjoo** [comment]
> LGTM.
> 
> Now, please review edge-tpu and mvnc and create unit-test for SNPE.

(After you've tested SNPE at Android Device first)

**Pull request merged**